source/slang/slang-ir.cpp

```diff
@@ -4325,11 +4327,114 @@ namespace Slang
         }
     }
 
+
+    void IRModule::trySearchForAndFillAllRayVariables()
+    {
+        //instead of forcing order of code, a single pass if needed can be used
+        //likley forcing order is a faster and better idea
+
+        if(!checkedForRayVariables)
+        {
+            checkedForRayVariables = true;
+            for(auto i : getGlobalInsts())
+            {
+                if(auto rayPayload = i->findDecoration<IRVulkanRayPayloadDecoration>())
```

Comment:
`findDecoration` is O(n), so instead of doing a bunch of findDecorations, iterate over all the decorations and check the type:
```
for (auto decor : i->getDecorations())
{
        if (auto rayPayload = as<IRVulkanRayPayloadDecoration>(decor))
               ...
        else if (...)
              ....
}
```